packages/controller/lib/setup/setupUpgrade.js

```diff
@@ -468,160 +451,149 @@ function Upgrade(options) {
             if (!forceDowngrade) {
                 try {
                     await checkDependencies(repoUrl[adapter].dependencies, repoUrl[adapter].globalDependencies);
-                } catch (e) {
-                    console.error(e.message);
-                    return typeof callback === 'function' && callback();
+                } catch (err) {
+                    return console.error(`Cannot check dependencies: ${err.message || err}`);
                 }
             }
 
             if (!forceDowngrade && (repoUrl[adapter].version === ioInstalled.common.version ||
                 tools.upToDate(repoUrl[adapter].version, ioInstalled.common.version))) {
-                console.log(`Adapter "${adapter}"${(adapter.length < 15) ? new Array(15 - adapter.length).join(' ') : ''} is up to date.`);
-                callback && callback();
+                return console.log(`Adapter "${adapter}"${(adapter.length < 15) ? new Array(15 - adapter.length).join(' ') : ''} is up to date.`);
             } else {
                 const targetVersion = version || repoUrl[adapter].version;
                 try {
                     if (!showUpgradeDialog(ioInstalled.common.version, targetVersion, adapter)) {
-                        console.log(`No upgrade of "${adapter}" desired.`);
-                        return typeof callback === 'function' && callback();
+                        return console.log(`No upgrade of "${adapter}" desired.`);
                     }
                 } catch (err) {
                     console.log(`Can not check version information to display upgrade infos: ${err.message}`);
                 }
 
                 console.log(`Update ${adapter} from @${ioInstalled.common.version} to @${targetVersion}`);
                 // Get the adapter from web site
-                install.downloadPacket(sources, `${adapter}@${targetVersion}`, null, (enableAdapterCallback, name, ioPack) =>
-                    finishUpgrade(name, ioPack, () => enableAdapterCallback(callback)));
+                const name = await install.downloadPacketAsync(sources, `${adapter}@${targetVersion}`);
```

Comment:
Is the "enableAdapterCallback logic" gone here?

Reply:
It is all now on one place in setupInstall.js:
![image](https://user-images.githubusercontent.com/4582016/144759055-49407c5c-8f4c-4c91-8092-bbea0b0bfd36.png)